trailmap/acquisition_paths.py

```diff
@@ -0,0 +1,42 @@
+import networkx as nx
+from pprint import pprint
+
+
+def conduct_apa(facility_dict_in, facility_dict_out):
+    ''' '''
+    G = build_graph(facility_dict_in, facility_dict_out)
+
+    sources = list(node for node, in_deg in G.in_degree() if in_deg == 0)
+    sinks = list(node for node, out_deg in G.out_degree() if out_deg == 0)
+
+    print("\nSimple paths")
+    pathways = find_simple_paths(G, sources, sinks)
+    print(pathways)
+
+    return G, pathways
+
+
+def build_graph(facility_dict_in, facility_dict_out):
+    '''Builds NetworkX graph from Cyclus input file
+    '''
+    G = nx.MultiDiGraph()
+    G.add_nodes_from(facility_dict_in.keys())
+    for receiver, incommods in facility_dict_in.items():
+        # find all facilities with that commod as an outcommod
+        for sender, outcommods in facility_dict_out.items():
+            connection = list(set(incommods).intersection(outcommods))
+            for commod in connection:
+                G.add_edge(sender, receiver, commodity=commod)
+    return G
+
+
+def find_simple_paths(G, sources, sinks):
+    ''' finds all simple paths between a given list of sources and targets
+    '''
+    pathways = set()
+    for source, sink in [(x, y) for x in sources for y in sinks]:
+        paths = list(nx.all_simple_paths(G, source=source, target=sink))
```

Comment:
Could this just be:
```suggestion
        pathways = set(nx.all_simple_paths(G, source=source, target=sink))
```
or
```suggestion
        pathways = set([tuple(path) for path in nx.all_simple_paths(G, source=source, target=sink)])
```

Is the tuple-ness important (I seem to remember that it is but can't remember why).

Reply:
you can't take the output of `nx.all_simple_paths(G, source=source, target=sink)` and put it into a set because the output is a list of lists, and you get an unhashable error because you're effectively trying to turn a list of lists into a set of lists (not allowed)

tuples for individual pathways are important because they allow elements to be used again within the tuple but don't allow duplicates of the entire tuple. Now I guess technically you don't need tuples for pathways when we're only caring about _simple paths_ because you won't get ever get cycles. However, when you get cycles, you want tuples as opposed to sets or lists

reference for variable names
pathways: all paths from any source to any sink in a fuel cycle
path: a single path from a source to a sink
paths: all paths from a single source to a single sink. If there is more than one source and/or sink, this is not all pathways through a fuel cycle

Reply:
@bam241 trying to consolidate all the conversation on this block of code on this comment thread

Reply:
might have some solution there: https://stackoverflow.com/questions/3523048/add-another-tuple-to-a-tuple-of-tuples